src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,576 @@
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the clusters to the DUT.
+        self.print_step(6, "Send Request Message to read one global attribute from all clusters on all endpoints")
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [Clusters.Objects.Descriptor.Attributes.AttributeList])
+        attribute_path_2 = AttributePath(
+        EndpointId = None,
+        ClusterId = None,
+        AttributeId = global_attribute_ids.GlobalAttributeIds.ATTRIBUTE_LIST_ID)
+
+        read_request = await self.default_controller.ReadAttribute(
+            self.dut_node_id,
+            [attribute_path_2]
+        )
+
+        for endpoint in read_request:
+            asserts.assert_in(Clusters.Objects.Descriptor, read_request[endpoint].keys(), "Descriptor cluster not in output")
+            asserts.assert_in(Clusters.Objects.Descriptor.Attributes.AttributeList, read_request[endpoint][Clusters.Objects.Descriptor], "AttributeList not in output")
+        
+        # Step 7
+        # TH sends the Read Request Message to the DUT to read all attributes from a cluster at all Endpoints
+        # AttributePath = [[Cluster = Specific ClusterID]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the Endpoints to the DUT.
+        self.print_step(7, "Send Request Message to read all attributes from one cluster at all endpoints")
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [Clusters.Objects.Descriptor])
+
+        for endpoint in read_request:
+            asserts.assert_in(Clusters.Objects.Descriptor, read_request[endpoint].keys(), "Descriptor cluster not in output")
+            asserts.assert_in(Clusters.Objects.Descriptor.Attributes.AttributeList, read_request[endpoint][Clusters.Objects.Descriptor], "AttributeList not in output")
+
+        # Step 8
+        # TH sends the Read Request Message to the DUT to read all attributes from all clusters at one Endpoint
+        # AttributePath = [[Endpoint = Specific Endpoint]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the Endpoints to the DUT.
+        self.print_step(8, "Send Request Message to read all attributes from all clusters at one endpoint")
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [0])
+
+        for endpoint in read_request:
+            asserts.assert_in(Clusters.Objects.Descriptor, read_request[endpoint].keys(), "Descriptor cluster not in output")
+            asserts.assert_in(Clusters.Objects.Descriptor.Attributes.ServerList, read_request[endpoint][Clusters.Objects.Descriptor], "ServerList not in output")
+            asserts.assert_equal(
+                read_request[endpoint][Clusters.Objects.Descriptor][Clusters.Objects.Descriptor.Attributes.ServerList],
```

Comment:
The check here should be that the server list of clusters should match the list of cluster IDs returned in the read. So if you do clusters = [c.id for c in read_request[endpoint].keys()] and remove the descriptor cluster, that should match the server list. Similarly, for each cluster, the attribute list should match the list of returned attributes.